filebeat/harvester/reader/encode.go

```diff
@@ -16,20 +16,24 @@ type Encode struct {
 // NewEncode creates a new Encode reader from input reader by applying
 // the given codec.
 func NewEncode(
-	in io.Reader,
+	in io.ReadCloser,
```

Comment:
just asking for io.Reader might be ok here. In stdlib sometimes this pattern is used:

```
// r is io.Reader
rc, ok := r.(io.ReadCloser)
if !ok {
  rc = ioutil.NopCloser(r)
}

// use rc here
```

That is, Close become optional. but using `io.ReadCloser` is ok too.